src/Chat/Bot/Misc/Hangman.hs

```diff
@@ -0,0 +1,16 @@
+module Chat.Bot.Misc.Hangman where
+
+
+data Hangman =
+    Hangman [Char]
+  deriving (Eq, Show)
+
+data HangmanResult =
+    HangmanRight
+  -- Contains the number of wrong guesses
+  | HangmanWrong Int
```

Comment:
FYI I've got two commits. I made the second one simpler so that keeping track of the string is done in `Run`, just to simply things. Not sure if that makes it _too_ easy or not?